module/VuFind/tests/unit-tests/src/VuFindTest/View/Helper/Root/MakeTagTest.php

```diff
@@ -0,0 +1,193 @@
+ */
+class MakeTagTest extends AbstractMakeTagTest
+{
+    /**
+     * Get makeTag helper with mock view
+     *
+     * return \Laminas\View\Helper\EscapeHtml
+     */
+    protected function getHelper()
+    {
+        $helper = new MakeTag();
+        $helper->setView($this->getViewWithHelpers());
+        return $helper;
+    }
+
+    /**
+     * Test that responds to common inputs
+     */
+    public function testAttributes()
+    {
+        $helper = $this->getHelper();
+
+        $this->assertEquals(
+            '<button class="btn" id="login">text</button>',
+            $helper('button', 'text', ['class' => 'btn', 'id' => 'login'])
+        );
+
+        // String
+        $this->assertEquals(
+            '<i class="btn">text</i>',
+            $helper('i', 'text', 'btn')
+        );
+
+        // Empty text
+        $this->assertEquals(
+            '<i class="fa&#x20;fa-awesome"></i>',
+            $helper('i', '', 'fa fa-awesome')
+        );
+
+        // Truthy attribute
+        $this->assertEquals(
+            '<a href="&#x2F;login" data-lightbox="1">Login</a>',
+            $helper('a', 'Login', ['href' => '/login', 'data-lightbox' => true])
+        );
+    }
+
+    /**
+     * Test escapeContent
+     */
+    public function testOptionEscape()
+    {
+        $helper = $this->getHelper();
+
+        $this->assertEquals(
+            '<button>This link is &lt;strong&gt;important&lt;/strong&gt;</button>',
+            $helper(
+                'button',
+                'This link is <strong>important</strong>',
+            )
+        );
+
+        $this->assertEquals(
+            '<button>This link is <strong>important</strong></button>',
+            $helper(
+                'button',
+                'This link is <strong>important</strong>',
+                [],
+                ['escapeContent' => false]
+            )
+        );
+
+        $this->assertEquals(
+            '<button>This link is &lt;strong&gt;important&lt;/strong&gt;</button>',
+            $helper(
+                'button',
+                'This link is <strong>important</strong>',
+                [],
+                ['escapeContent' => true]
+            )
+        );
+    }
+
+    /**
+     * Test escapeContent
+     */
+    public function testVoidElements()
+    {
+        $helper = $this->getHelper();
+
+        $this->assertEquals(
+            '<img src="book.gif" />',
+            $helper(
+                'img',
+                '',
+                ['src' => 'book.gif']
+            )
+        );
+
+        // Class only
+        $this->assertEquals(
+            '<br class="sm&#x3A;hidden" />',
+            $helper(
+                'br',
+                '',
+                'sm:hidden'
+            )
+        );
+    }
+
+    /**
+     * Test tag name edge cases
+     */
+    public function testValidTagNames()
+    {
+        $helper = $this->getHelper();
+
+        $helper('CAPITAL', '');
```

Comment:
You don't seem to have any assertions in this test... shouldn't you be checking the return values of these calls?

Reply:
Because the helper throws an error on invalid tests, this test is just making sure that no errors throw. I can add comments to this effect unless there's a more explicit way to test this.

Reply:
The helper should be returning some HTML, so why not `$this->assertEquals()` appropriate values to be sure we are getting the expected output?